fields/ModuleNavigationField.php

```diff
@@ -61,10 +61,9 @@ protected function renderStatic($entry)
             }
 
             $disabledLink = $this->courseManager->isModuleDisabledLink($learner, $course, $module);
-            // TODO implement getNextActivity for a learner, for the moment choose the first activity of the module
-            $labelStart = $learner && $learner->isAdmin() && $module->getStatus($course) != ModuleStatus::OPEN ?
-                _t('LMS_BEGIN_ONLY_ADMIN')
-                : _t('LMS_BEGIN');
+            $tmpData = $this->courseManager->getLastAccessibleActivityTagAndLabelForLearner($learner, $course, $module) ;
+            $nextActivityTag = $tmpData['tag'];
+            $labelStart = $tmpData['label'];
             if ($module->isAccessibleBy($learner, $course)) {
```

Comment:
c'est ici que tu as fais différer, et c'est pas ici que les statuts se calculent : c'est au niveau de $module->getStatus(...) qu'il faut intégrer ce test.

Reply:
le module possède un statut qui DOIT être indépendant du learner, .....sinon ça veut dire que le comportement du module varie en fonction des learners (aïe souci à prévoir).
Donc normalement NOT_ACCESSIBLE ne devrait pas être possible dans module->status.
je serai plutôt d'avis de déporter isModuleAccessibleBy dans learnerManager et de gérer le statut à ce niveau.

Reply:
on aura le temps d'en discuter plus généralement à la réunion mais voici ce que disent les commentaires du getStatus :

```
/**
  * Check if given module is accessible and open for current user, for a given course.
  * Admins may navigate throught closed modules.
  ...
  ```
  le "for current user" laisse bien à supposer que  ça dépend du status. Il y a aussi :
  ```
  // TODO finish the scenarisation
 $this->status = ModuleStatus::OPEN;
```
qui était assez clair qu'on voyait bien la scénarisation ici (je dis on car à la base c'est flo qui avait écris cette fonction).

Je comprends toutefois mieux maintenant où se situent tes interrogations par rapport aux choix qui ont été fait dans l'architecture.
Dans ce cas, on a bien un statut calculé (car le seul attribut qui est stockée dans la fiche c'est si l'activité est active ou non), et on va pouvoir rajouter la scénarisation en s'appuyant indirectement sur des méthodes comme isModuleAccessibleBy définit dans le learnerManager.

Je dis indirectement car ça me paraît intéressant de garder le $module->isAccessibleBy pour d'autres raisons (on en rediscutera).